src/components/List/List.tsx

```diff
@@ -2,29 +2,33 @@ import * as _ from 'lodash'
 import * as React from 'react'
 import * as PropTypes from 'prop-types'
 
-import { customPropTypes, UIComponent, childrenExist } from '../../lib'
+import { customPropTypes, childrenExist, UIComponent } from '../../lib'
 import ListItem from './ListItem'
 import { listBehavior } from '../../lib/accessibility'
-import { Accessibility } from '../../lib/accessibility/interfaces'
+import { Accessibility, AccessibilityActionHandlers } from '../../lib/accessibility/interfaces'
+import {
+  ContainerFocusHandler,
+  IFocusContainerProps,
+  IFocusContainerState,
+} from '../../lib/accessibility/FocusHandling/FocusContainer'
 
 import { ComponentVariablesInput, ComponentPartStyle } from '../../../types/theme'
 import { Extendable, ReactChildren, ItemShorthand } from '../../../types/utils'
 
-export interface IListProps {
+export interface IListProps extends IFocusContainerProps<ItemShorthand> {
   accessibility?: Accessibility
   as?: any
   children?: ReactChildren
   className?: string
   debug?: boolean
-  items?: ItemShorthand[]
   selection?: boolean
   truncateContent?: boolean
   truncateHeader?: boolean
   styles?: ComponentPartStyle
   variables?: ComponentVariablesInput
 }
 
-class List extends UIComponent<Extendable<IListProps>, any> {
+class List extends UIComponent<Extendable<IListProps>, IFocusContainerState> {
```

Comment:
one observation is that, formally, we don't need to translate state back from the `FocusContainer` to component. Essentially, we could keep this state being encapsulated in `FocusContainer`, and the only thing that we would need is some notification mechanism to let component note that it should update. We could further elaborate on this idea to make `FocusContainer` being framework agnostic